gcp-function-cloudevents/src/main/java/io/micronaut/gcp/function/cloudevents/CloudEventTypesSerde.java

```diff
@@ -0,0 +1,169 @@
+
+/**
+ * @see <a href="https://github.com/googleapis/google-cloudevents-java">Google Cloud Events</a>.
+ * @author Sergio del Amo
+ * @since 4.7.0
+ */
+@SerdeImport(ReferenceEventData.class)
+@SerdeImport(com.google.events.cloud.audit.v1.Metadata.class)
+@SerdeImport(com.google.events.firebase.auth.v1.Metadata.class)
+@SerdeImport(AuthEventData.class)
+@SerdeImport(UpdateType.class)
+@SerdeImport(UpdateOrigin.class)
+@SerdeImport(RemoteConfigEventData.class)
+@SerdeImport(AnalyticsLogData.class)
+@SerdeImport(AppInfo.class)
+@SerdeImport(UserDim.class)
+@SerdeImport(AnalyticsValue.class)
+@SerdeImport(BundleInfo.class)
+@SerdeImport(DeviceInfo.class)
+@SerdeImport(TrafficSource.class)
+@SerdeImport(com.google.events.firebase.analytics.v1.Value.class)
+@SerdeImport(com.google.events.cloud.firestore.v1.Value.class)
+@SerdeImport(LtvInfo.class)
+@SerdeImport(GeoInfo.class)
+@SerdeImport(OldValue.class)
+@SerdeImport(GeoPointValue.class)
+@SerdeImport(OldValueField.class)
+@SerdeImport(DocumentEventData.class)
+@SerdeImport(MapValue.class)
+@SerdeImport(NullValue.class)
+@SerdeImport(MapValueField.class)
+@SerdeImport(ValueElement.class)
+@SerdeImport(ArrayValue.class)
+@SerdeImport(Artifacts.class)
+@SerdeImport(ObjectsTiming.class)
+@SerdeImport(BuildEventData.class)
+@SerdeImport(com.google.events.cloud.audit.v1.Status.class)
+@SerdeImport(com.google.events.cloud.cloudbuild.v1.Status.class)
+@SerdeImport(ResolvedRepoSourceClass.class)
+@SerdeImport(Objects.class)
+@SerdeImport(Type.class)
+@SerdeImport(PushTiming.class)
+@SerdeImport(Secret.class)
+@SerdeImport(LogStreamingOption.class)
+@SerdeImport(MachineType.class)
+@SerdeImport(RequestedVerifyOption.class)
+@SerdeImport(StorageSourceClass.class)
+@SerdeImport(TimeSpan.class)
+@SerdeImport(ArtifactTiming.class)
+@SerdeImport(Results.class)
+@SerdeImport(SourceProvenance.class)
+@SerdeImport(PullTiming.class)
+@SerdeImport(Options.class)
+@SerdeImport(Source.class)
+@SerdeImport(RepoSourceClass.class)
+@SerdeImport(SourceProvenanceHash.class)
+@SerdeImport(SubstitutionOption.class)
+@SerdeImport(ResolvedStorageSourceClass.class)
+@SerdeImport(Volume.class)
+@SerdeImport(Logging.class)
+@SerdeImport(StepTiming.class)
+@SerdeImport(SchedulerJobData.class)
+@SerdeImport(StorageObjectData.class)
+@SerdeImport(CustomerEncryption.class)
+@SerdeImport(AuthorizationInfo.class)
+@SerdeImport(ProtoPayload.class)
+@SerdeImport(ResourceLocation.class)
+@SerdeImport(Auth.class)
+@SerdeImport(Operation.class)
+@SerdeImport(DestinationAttributes.class)
+@SerdeImport(Severity.class)
+@SerdeImport(FirstPartyPrincipal.class)
+@SerdeImport(RequestMetadata.class)
+@SerdeImport(Detail.class)
+@SerdeImport(LogEntryData.class)
+@SerdeImport(ServiceAccountDelegationInfo.class)
+@SerdeImport(Request.class)
+@SerdeImport(AuthenticationInfo.class)
+@SerdeImport(Message.class)
+@SerdeImport(MessagePublishedData.class)
+@SerdeImport(CustomerEncryption.class)
+public class CloudEventTypesSerde {
```

Comment:
I would move this to the serialization module since it seems generally useful

Reply:
add this class with a `compileOnly` dependency to     api(libs.managed.google.cloudevent.types)
in micronaut-serialization repo ? 

@guillermocalvo could you create a PR in serialisation with this class and a ObjectStorageData test.

Reply:
@sdelamo @graemerocher https://github.com/micronaut-projects/micronaut-serialization/pull/324

Reply:
no that is is not what I meant, it would need to go into a new module like `serde-gcp-cloudevents`

Reply:
@guillermocalvo I talked to @graemerocher  

We are going to keep all the code in the GCP repository but in two modules. 

gcp-serde-cloudevents
gcp-function-cloudevents

Can you update this PR?